src/common/transformations/src/transformations/common_optimizations/remove_multi_subgraph_op_dangling_params.cpp

```diff
@@ -178,19 +178,21 @@ bool ov::pass::RemoveMultiSubGraphOpDanglingParamsResults::run_on_model(const st
                     if (std::count(std::begin(to_remove_descriptors_indexes[body_idx]),
                                    std::end(to_remove_descriptors_indexes[body_idx]),
                                    desc_idx) > 0) {
-                        auto& body_param = body_params[body_in_descriptors[desc_idx]->m_body_parameter_index];
-                        body_func->remove_parameter(body_param);
+                        if (body_in_descriptors[desc_idx]->m_body_parameter_index < body_params.size()) {
```

Comment:
usually when "body_in_descriptors[desc_idx]->m_body_parameter_index" > body_params.size()
it means that TI/Loop/If op is not consistent?
is this a real scenario occurring in some model or only in a test?

do we know the root cause of this problem? what causes this situation to arise?

Reply:
If here were used `at` instead `[]` it will fail on linux also.
The `body_func->remove_parameter(body_param);`  change number of inputs and `body_in_descriptors[desc_idx]->m_body_parameter_index` still index value as before parameter removal.

This new version works exactly as previous one just not try access data which not exists (undefined state). As parameter removal will not completely remove data there is access to something which looks like is meaning less and there is no failure. The MSVC build just detects this incorrect access.

Maybe as next step this part should be improved to update descriptor's indexes when parameter removed and `at` or `[]` can be used without checks.

Reply:
>> The body_func->remove_parameter(body_param); change number of inputs and body_in_descriptors[desc_idx]->m_body_parameter_index still index value as before parameter removal.

this is clear.
 but `update_body_param_desc` call ( probably we should move  `update_body_param_desc` call to this `if`.) have to update `m_body_parameter_index`  and the descriptors have to be in the correct state after this call and point to the correct indexes.  So I expect that the situation (undefined state) will never happen.

the logic of the transformation looks like that:
remove param -> update the param index in the desc -> remove input -> update the input index in the desc

Reply:
It looks like `update_body_param_desc` can be not correctly implemented when there are multiple descriptors pointing to same input.
In case where two descriptors pointing to index `1` on first the parameter is removed, then descriptors updated  (it upadte indexes by `-1`, all above but  current) in next step there is another descriptor, which want access parameter at index `1` (but it was removed and exception can be thrown).

I can move this call inside `if`.